### PR TITLE
Remove hook addLogEntry and its usages

### PR DESCRIPTION
**Pull request by bytehead**
See #4513

`contao/docs` needs to be adjusted as well.

### CODE REVIEW

File: core-bundle/tests/Monolog/ContaoTableHandlerTest.php [hunk 1]
```diff
@@ -43,13 +42,8 @@ public function testSupportsReadingAndWritingTheDbalServiceName(): void
         $this->assertSame('foobar', $handler->getDbalServiceName());
     }
 
-    /**
```
Comment (Toflar) [inline core-bundle/tests/Monolog/ContaoTableHandlerTest.php:46]: Imho the `tearDown()` can also be removed in this method now.

Comment (bytehead) [reply core-bundle/tests/Monolog/ContaoTableHandlerTest.php:46]: Good catch!

Comment (bytehead) [reply core-bundle/tests/Monolog/ContaoTableHandlerTest.php:46]: Removed in 52b5accdd.

**Review by Toflar — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Pull request merged**